### PR TITLE
[2022-10-27] Roman to Integer Swift

### PR DESCRIPTION
**Pull request by saranghe41**
## Roman to Integer
Roman numerals are represented by seven different symbols: I, V, X, L, C, D and M.

Symbol       Value
I             1
V             5
X             10
L             50
C             100
D             500
M             1000
For example, 2 is written as II in Roman numeral, just two ones added together. 12 is written as XII, which is simply X + II. The number 27 is written as XXVII, which is XX + V + II.

Roman numerals are usually written largest to smallest from left to right. However, the numeral for four is not IIII. Instead, the number four is written as IV. Because the one is before the five we subtract it making four. The same principle applies to the number nine, which is written as IX. There are six instances where subtraction is used:

I can be placed before V (5) and X (10) to make 4 and 9. 
X can be placed before L (50) and C (100) to make 40 and 90. 
C can be placed before D (500) and M (1000) to make 400 and 900.
Given a roman numeral, convert it to an integer.

## Time Complex & Space Complex
Runtime: 71 ms, faster than 11.73% of Swift online submissions for Roman to Integer.
Memory Usage: 15.5 MB, less than 5.45% of Swift online submissions for Roman to Integer.

## 참고 링크
* [Leet Code](https://leetcode.com/problems/roman-to-integer/)

## 기타 사항
다양한 방면으로 생각하려고 노력해야할 것 같다🧐

### CODE REVIEW

File: Roman to Integer/Roman_to_Integer.swift [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        let romanToIntDics: [Character:Int] = ["I":1, "V":5, "X":10, "L":50, "C":100, "D":500, "M":1000]
+        var changeS = s.replacingOccurrences(of: "IV", with: "IIII")
+            .replacingOccurrences(of: "IX", with: "VIIII")
+            .replacingOccurrences(of: "XL", with: "XXXX")
+            .replacingOccurrences(of: "XC", with: "LXXXX")
+            .replacingOccurrences(of: "CD", with: "CCCC")
+            .replacingOccurrences(of: "CM", with: "DCCCC")
+        var sum = 0
+        
+        for ch in changeS {
```
Comment (kyu1204) [inline Roman to Integer/Roman_to_Integer.swift:14]: 이 부분 swift 에는 혹시 sum() 이나 map() 같은 iterator 없나요?
있으면 좀 더 코드를 간결하게 만들 수 있을 것 같아요

**Review by kyu1204 — APPROVED** [review]
son
수고하셨습니다~ 코멘트 하나 남겼어요 확인 부탁합니다.

File: Roman to Integer/Roman_to_Integer.swift (hunk 1 above)
Comment (saranghe41) [reply Roman to Integer/Roman_to_Integer.swift:14]: 참고해서 찾아보니 map() 고차함수를 사용할 수는 있을 것 같은데, map()의 return이 배열이라 코드가 더 길어지지 않을까 싶네요🧐

Comment (kyu1204) [reply Roman to Integer/Roman_to_Integer.swift:14]:
```suggestion
        var sum = changeS.map({ (ch: Character) -> Int in return romanToIntDics[ch]! }).reduce(0, +)
```
이런식으로 작성할 수 있을 것 같네요

**Review by kyu1204 — APPROVED** [review]

File: Roman to Integer/Roman_to_Integer.swift (hunk 1 above)
Comment (saranghe41) [reply Roman to Integer/Roman_to_Integer.swift:14]: Wow.. 되는군요👏

File: Roman to Integer/Roman_to_Integer.swift [hunk 2]
```diff
@@ -11,7 +11,7 @@ class Solution {
             .replacingOccurrences(of: "CM", with: "DCCCC")
         var sum = 0
```
Comment (kyu1204) [inline Roman to Integer/Roman_to_Integer.swift:12]: 이 코드는 없어져도 될것 같아요!

File: Roman to Integer/Roman_to_Integer.swift [hunk 3]
```diff
@@ -11,7 +11,7 @@ class Solution {
...
         
-        for ch in changeS {
+        var sum = changeS.map({ (ch: Character) -> Int in return romanToIntDics[ch]! }).reduce(0, +)
             sum += romanToIntDics[ch]!
```
Comment (kyu1204) [inline Roman to Integer/Roman_to_Integer.swift:15]: 여기 밑에도 남아있습니다~!

**Review by kyu1204 — COMMENTED** [review]
마지막 불필요한 코드정리 한번 부탁드려요~

**Pull request merged**